actions/lib/supr_utils.py

```diff
@@ -0,0 +1,113 @@
+import requests
+import json
+import math
+from datetime import date
+from dateutil.relativedelta import relativedelta
+
+# Needs to be run in a Stackstorm virtualenv
+from st2actions.runners.pythonrunner import Action
+
+
+class SuprUtils:
+
+    DATE_FORMAT = '%Y-%m-%d'
+
+    @staticmethod
+    def search_by_email(base_url, email, user, key):
+        search_person_url = '{}/person/search/'.format(base_url)
+        # Search case insensitive
+        params = {'email_i': email}
+        response = requests.get(search_person_url, params=params, auth=(user, key))
+
+        if response.status_code != 200:
+            raise AssertionError("Status code returned when trying to get PI id for email: "
```

Comment:
same comment as above.. also, this case is not handled in the action, right? not sure what is the right thing to do though, perhaps re-raise the exception in the calling context but on the other hand, that will stop the workflow. Perhaps silently proceeding to the next project is the best way to handle this..

Reply:
I would say silently moving on to process the rest of the projects. If the infrastructure glitches during a run, it hopefully works next time it's executed. My impression is that it probably runs more than needed anyway.